### PR TITLE
Use process.env to allow configuring of socket permissions

### PR DESCRIPTION
**Pull request by gomasy**
Puma can be set with environment variables, but streaming cannot do that...

### CODE REVIEW

**Comment by Gargron** [comment]
>Puma can be set with environment variables, but streaming cannot do that...

I cannot find a Puma setting for `SOCKET_PERMISSION`, could you clarify?

**Comment by Gomasy** [comment]
screenshot from 2018-10-03 14-27-33
https://github.com/puma/puma/blob/master/README.md

Puma can change permissions of the socket by adding the umask parameter.

File: streaming/index.js [hunk 1]
```diff
@@ -586,7 +586,7 @@ const startWorker = (workerId) => {
 
   if (process.env.SOCKET || process.env.PORT && isNaN(+process.env.PORT)) {
     server.listen(process.env.SOCKET || process.env.PORT, () => {
-      fs.chmodSync(server.address(), 0o666);
+      fs.chmodSync(server.address(), process.env.SOCKET_PERMISSION || 0o666);
```
Comment (Gargron) [inline streaming/index.js]: SOCKET_PERMISSION will be a string, but 0o666 is not a string. Perhaps conversion is needed here?

Also maybe call it SOCKET_UMASK?

**Review by Gargron — CHANGES_REQUESTED** [review]

**Comment by Gomasy** [comment]
@Gargron It has changed.
It is calculated if exists, but if it does not exist 0o666 is set as usual.

**Pull request closed without merging**